src/debug.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Tlf - contest logging program for amateur radio operators
+ * Copyright (C) 2025 Thomas Beierlein <tb@forth-ev.de>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+
+#include "debug.h"
+#include "get_time.h"
+#include "globalvars.h"
+#include "pthread.h"
+#include "stdbool.h"
+
+/* check if any debug level is active */
+bool debug_is_active() {
+    return (debuglevel > 0);
+}
+
+bool debug_init() {
+    char debugbuffer[80];
+
+    if (debug_is_active()) {
+	/* write timestamp as start entry into debug log */
+	FILE *fp = fopen(DEBUG_LOG, "a");
+	if (fp == NULL) {
+	    return FALSE;
+	}
+
+	format_time(debugbuffer, sizeof(debugbuffer),
+		    "\nStarted %d/%m/%Y %H:%M\n");
+	fputs(debugbuffer, fp);
+	fclose(fp);
+    }
+    return TRUE;
+}
+
+/* check if message needs to be logged */
+static bool needs_logging(enum debuglevel lvl) {
+    return (lvl <= debuglevel);
+}
+
+void debug_log(enum debuglevel lvl,
+	       const char *fmt,
+	       ...) {
+
+    static pthread_mutex_t debug_mutex = PTHREAD_MUTEX_INITIALIZER;
+    char debugbuffer[160];
+    va_list args;
+
+
+    if (!needs_logging(lvl)) {
+	return;
+    }
+
+    pthread_mutex_lock(&debug_mutex);
+
+    FILE *fp = fopen(DEBUG_LOG, "a");
+    if (fp == NULL) {	    /* ignore failure to write debug log */
+	pthread_mutex_unlock(&debug_mutex);
+	return;	    /* to not disturb logging activity */
+    }
+
+    format_time(debugbuffer, sizeof(debugbuffer), "%H:%M:%S ");
```

Comment:
let's also output ms, as sub-second resolution can come handy sometimes.
here's my makeshift solution
```
char   timestamp[24];
struct timeval current_time; 
gettimeofday(&current_time, NULL); 
strftime(timestamp, sizeof(timestamp), "%F %T", localtime(&current_time.tv_sec)); 
char ts[100]; 
sprintf(ts, "%s.%03ld", timestamp, current_time.tv_usec/1000);
```

(I guess format_time uses UTC which could be confusing)

Reply:
Hmm, keeping UTC would keep the log and the debug output in sync. Not sure here.

Reply:
I see. Debug log is more meant for a human, hence is my preference for local time. But at least let's add UTC or Z to make it unambiguous.

Reply:
As ham radio operator most of us are familiar with times in UTC. But I agree that we should state it explicitly.

To save line space I would still abstain from having the date in each line (9 characters) and also of the 'Z' or 'UTC' on each line.

At the moment logging starts with a comment 'Started 13/02/2025 21:10' which shows the actual date. I would suggest to change it into 'Started 13/02/2025 21:10 (All times in UTC)' to make it clear. How about that?

Reply:
My preference for timestamp format is ISO (2025-01-30T17:07:48Z) as this is well defined, has natural ordering and does not have the ambiguity of 05/02/2025 vs. 02/05/2025.
T and Z could be dropped. For the start something like 'Started 2025-01-30 17:07:48,123 UTC' would be fine.

This means that the start line is different from the bulk of the log. Normally logs a homogeneous in this sense, each line has the same structure.

Reply:
I see the point about 2025-01-30 instead of 30/01/2025.

Looking through the code we already use '30 Jan 25' or '20250130' (in `do_backup()`). 
I wonder if we really want to introduce a third format (that questions also my format in these PR), choose one of the other already used or move the others to the one proposed by you (not possible for QSO data in logfile)?

Reply:
20250130 is also fine with me for debug logging.

Reply:
See last commits.